### PR TITLE
Clarify drop_vars return value.

### PR DESCRIPTION
**Pull request by rpgoldman**
The previous documentation was not clear about whether the variable
dropping was "inplace" or created a fresh Dataset.

### CODE REVIEW

File: xarray/core/dataarray.py [hunk 1]
```diff
@@ -1942,7 +1942,7 @@ def drop_vars(
         Returns
         -------
         dropped : Dataset
-
+            New Dataset copied from ``self`` with variables removed.
```
Comment (max-sixty) [inline xarray/core/dataarray.py:1945]:
```suggestion
            New Dataset copied from `self` with variables removed.
```

File: xarray/core/dataarray.py [hunk 2]
```diff
@@ -1928,7 +1928,7 @@ def T(self) -> "DataArray":
     def drop_vars(
         self, names: Union[Hashable, Iterable[Hashable]], *, errors: str = "raise"
     ) -> "DataArray":
-        """Drop variables from this DataArray.
+        """Drop variables from this DataArray. *Not* done inplace.
```
Comment (max-sixty) [inline xarray/core/dataarray.py:1931]:
```suggestion
        """Returns an array with dropped variables.
```

**Comment by max-sixty** [comment]
Thanks @rpgoldman !

What do you think of that change? Feel free to discard if you disagree and we'll merge your initial suggestion.

**Comment by rpgoldman** [comment]
> Thanks @rpgoldman !
> 
> What do you think of that change? Feel free to discard if you disagree and we'll merge your initial suggestion.

How about we keep your sentence which is _much_ better than my first sentence, _and_ my second sentence? I think it's important to make the point that this function does not have side effects.

Any idea why the docs build failed?

**Comment by keewis** [comment]
> Any idea why the docs build failed?

that has been fixed in #4254 so merging `master` should be enough.

**Comment by max-sixty** [comment]
I realize that we may have lost a sentence through the merges; maybe someone clicked on apply suggestion to apply my suggestion. To the extent that wasn't your intention @rpgoldman lmk and I'll add back

**Pull request merged**